types/vue.d.ts

```diff
@@ -39,10 +41,9 @@ export declare class Vue {
   $emit(event: string, ...args: any[]): this;
   $nextTick(callback?: (this: this) => void): void;
   $createElement(
-    tag?: string | Vue,
+    tag?: string | Component | (() => string | Component),
```

Comment:
Actually the function type of this argument is the factory of async component.
So I think following type is more appropriate.

```js
string |
Component |
((
  resolve: (component: Component) => void,
  reject: any
) => Promise<Component> | Component | void)
```

Maybe it would be better to define `AsyncComponent` for this function type.